### PR TITLE
Add python test source files to project

### PR DESCRIPTION
**Pull request by domi4484**
Otherwise they are not available in project search or locator bar (at least in QtCreator version 6.0.0)

### CODE REVIEW

File: tests/src/python/CMakeLists.txt [hunk 1]
```diff
@@ -2,6 +2,9 @@ set (ENABLE_MSSQLTEST FALSE CACHE BOOL "Enable MSSQL provider tests")
 set (ENABLE_ORACLETEST FALSE CACHE BOOL "Enable Oracle provider tests")
 set (ENABLE_HANATEST FALSE CACHE BOOL "Enable HANA provider tests")
 
+FILE(GLOB_RECURSE PythonTestsFiles "*.py")
+add_custom_target(python_tests SOURCES ${PythonTestsFiles})
```
Comment (3nids) [inline tests/src/python/CMakeLists.txt:6]: what about creating a custom target directly in [ADD_PYTHON_TEST](https://github.com/qgis/QGIS/blob/master/cmake/UsePythonTest.cmake)?
The command would something like `$PYTHONPATH $SOURCE`.
We would have an executable target directly accessible from the IDE.

Comment (domi4484) [reply tests/src/python/CMakeLists.txt:6]: Yes that would be much better thanks

**Comment by domi4484** [comment]
@3nids with `add_executable` I have a runnable entry for each test, but I can't get them to just run, he tries to compile them... I also tried with `add_custom_target` without success...
I have very limited knowledge of cmake, do you have any idea how i could do that?

**Comment by m-kuhn** [comment]
I am not sure how this is meant to work:

- `add_executable` is meant for "executables built from source" which I don't think applies here
- `add_custom_target` can have commands specified, which could be a `python` command with the test as a parameter. However, this command will be for _build_ time, not for _runtime_, so I don't think it will be added there

**Comment by domi4484** [comment]
Yes ideally there should be something like `add_custom_executable`

**Comment by 3nids** [comment]
I have created another PR to simplify a bit the macro.
see https://github.com/qgis/QGIS/pull/46642

I am not sure if we can have a target for each of the python tests (to easily run them from the IDE, at the cost of having 400 additional targets).

Anway, for now, I would suggest to add a single target for all the python tests as [you originally suggested](https://github.com/qgis/QGIS/pull/46583/commits/58e573f898a35a14d216005a9f88a0461d3d8753#diff-a0f2d85a07d8edc1df7c7ceb8e41dbc1937b52ccfd8cb4bd5f2c6bf190ab2752R6) to have the Python files in the sources.

**Pull request closed without merging**